hack/update-deepcopy.sh

```diff
@@ -2,17 +2,28 @@
 
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../../k8s.io/code-generator)}
+# Build deepcopy-gen when it's not present and not overriden for a specific file.
+if [ -z "${DEEPCOPY_GEN:-}" ];then
+  ${TOOLS_MAKE} deepcopy-gen
+  DEEPCOPY_GEN="${TOOLS_OUTPUT}/deepcopy-gen"
+fi
 
 # API_GROUP_VERSION_PACKAGES is a string of <group>/<version>.
 # The deepcopy gen needs a string of <group>:<version> so substitute slashes for colons.
 inputArg="$(echo ${API_GROUP_VERSIONS//\//:})"
 
 verify="${VERIFY:-}"
 
-GOFLAGS="" bash ${CODEGEN_PKG}/generate-groups.sh "deepcopy" \
-  github.com/openshift/api/generated \
-  github.com/openshift/api \
-  "${inputArg}" \
-  --go-header-file ${SCRIPT_ROOT}/hack/empty.txt \
+# API_GROUP_VERSION_PACKAGES is a string of <group>/<version>.
+# The deepcopy gen needs a comma-separated list of ./<group>/<version> so print in that format and remove the leading comma.
+inputArg="$(printf ",./%s" ${API_GROUP_VERSIONS})"
+inputArg="${inputArg:1}"
+
+echo Generating Deepcopy for ${API_GROUP_VERSIONS}
+
+GOPATH= ${DEEPCOPY_GEN} \
```

Comment:
nit: You should quote `"${DEEPCOPY_GEN}"` here.

In practise it's only going to break anybody with whitespace in their `${SCRIPT_ROOT}`, so possibly nobody, but it's best practise.